src/jit-format/jit-format.cs

```diff
@@ -0,0 +1,682 @@
+                        _fix = (found) ? fix : _fix;
+                    }
+                }
+                else
+                {
+                    Console.Error.WriteLine("Can't find format.json on {0}", _jitUtilsRoot);
+                }
+            }
+            else
+            {
+                Console.WriteLine("Environment variable JIT_FORMAT_ROOT not found - no configuration loaded.");
+            }
+        }
+
+        public static CommandResult TryCommand (string name, IEnumerable<string> commandArgs, bool capture = false)
+        {
+            try 
+            {
+                Command command =  Command.Create(name, commandArgs);
+
+                if (capture)
+                {
+                    // Capture stdout/stderr for consumption within tool.
+                    command.CaptureStdOut();
+                    command.CaptureStdErr();
+                }
+                else
+                {
+                    // Wireup stdout/stderr so we can see output.
+                    command.ForwardStdOut();
+                    command.ForwardStdErr();
+                }
+
+                return command.Execute();
+            }
+            catch (CommandUnknownException e)
+            {
+                Console.Error.WriteLine("\nError: {0} command not found!  Add {0} to the path.", name, e);
+                Environment.Exit(-1);
+                return CommandResult.Empty;
+            }
+        }
+
+        public T ExtractDefault<T>(string name, out bool found)
+        {
+            var token = _jObj["default"][name];
+
+            if (token != null)
+            {
+                found = true;
+
+                try
+                {
+                    return token.Value<T>();
+                }
+                catch (System.FormatException e)
+                {
+                    Console.Error.WriteLine("Bad format for default {0}.  See format.json", name, e);
+                }
+            }
+
+            found = false;
+            return default(T);
+        }
+
+        public bool IsWindows { get { return (_os == "Windows_NT"); } }
+        public bool DoVerboseOutput { get { return _verbose; } }
+        public bool DoClangTidy { get { return !_untidy; } }
+        public bool DoClangFormat { get { return !_noformat; } }
+        public bool Fix { get { return _fix; } }
+        public bool BuildCompileCommands { get { return _buildCompileCommands; } }
+        public string CoreCLRRoot { get { return _rootPath; } }
+        public string BuildLog { get { return _buildLog; } }
+        public string CompileCommands { get { return _compileCommands; } }
+        public IReadOnlyList<string> Filenames { get { return _filenames; } }
+        public IReadOnlyList<string> Projects { get { return _projects.Count == 0 ? new List<string>{"dll"} : _projects; } }
+        public string SourceDirectory { get { return _srcDirectory; } }
+    }
+
+    public class jitformat
```

Comment:
For the main class, though odd, it was done to match the commandline on linux (case sensitive).  I thought it a special case.